### PR TITLE
Add an init container to fix templates repo directory ownership

### PR DESCRIPTION
**Pull request by GeorgianaElena**
Fixes https://github.com/2i2c-org/infrastructure/issues/1695  (more context about these changes, in the issue)

### CODE REVIEW

**Review by sgibson91 — APPROVED** [review]
This LGTM, but please reserve the right to run it by someone other than myself for whether it is optimal or not!

**Comment by GeorgianaElena** [comment]
@yuvipanda, do you also want to take a look at this one, please? I plan to merge this tomorrow evening if no objections are raised 🚀

**Comment by GeorgianaElena** [comment]
Merging this now since it's got one approval and no objections. Thank you!

**Pull request merged**
